pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,568 @@
+            - `q1deg + q2deg + q3deg = feat_deg` for each combination, where `min_deg <= feat_deg <= deg`.
+    """
+    fit_degs = []
+    for feat_deg in range(min_deg, deg + 1):
+        max_deg = feat_deg - 3
+        if max_deg < 0:
+            continue
+        possible_occupations = _generate_bin_occupations(max_deg, 3)
+        for occ in possible_occupations:
+            q1deg = 1 + occ[0]
+            q2deg = 1 + occ[1]
+            q3deg = 1 + occ[2]
+            fit_degs.append((q1deg, q2deg, q3deg))
+
+    return fit_degs
+
+
+def _fit_threebody(threemode_op, deg, min_deg=3):
+    r"""Fits the three-mode operator to get three-body coefficients.
+
+    Args:
+        threemode_op (TensorLike[float]): threemode operator
+        deg (int): maximum degree of taylor form polynomial
+        min_deg (int): minimum degree of taylor form polynomial
+
+    Returns:
+        tuple (list(list(float)), list(list(float))):
+            - the three-body coefficients
+            - the predicted three-body PES using fitted coefficients
+    """
+    _import_sklearn()
+    from sklearn.linear_model import LinearRegression
+
+    nmodes, _, _, quad_order, _, _ = np.shape(threemode_op)
+    gauss_grid, _ = np.polynomial.hermite.hermgauss(quad_order)
+
+    if deg < min_deg:
+        raise Exception(
+            f"Taylor expansion degree is {deg}<{min_deg}, minimal degree is set by min_deg keyword!"
+        )
+
+    predicted_3D = np.zeros_like(threemode_op)
+    fit_degs = _threebody_degs(deg)
+    num_coeffs = len(fit_degs)
+    coeffs = np.zeros((nmodes, nmodes, nmodes, num_coeffs))
+
+    grid_3D = np.array(np.meshgrid(gauss_grid, gauss_grid, gauss_grid))
+    q1 = grid_3D[0, ::].flatten()
+    q2 = grid_3D[1, ::].flatten()
+    q3 = grid_3D[2, ::].flatten()
+    idx_3D = np.array(np.meshgrid(range(quad_order), range(quad_order), range(quad_order)))
+    idx1 = idx_3D[0, ::].flatten()
+    idx2 = idx_3D[1, ::].flatten()
+    idx3 = idx_3D[2, ::].flatten()
+    num_3D = len(q1)
+
+    features = np.zeros((num_3D, num_coeffs))
+    for deg_idx, Qs in enumerate(fit_degs):
+        q1deg, q2deg, q3deg = Qs
+        features[:, deg_idx] = q1 ** (q1deg) * q2 ** (q2deg) * q3 ** (q3deg)
+
+    for i1 in range(nmodes):
+        for i2 in range(i1):
+            for i3 in range(i2):
+                Y = threemode_op[i1, i2, i3, idx1, idx2, idx3]
+
+                poly3D_reg_model = LinearRegression()
+                poly3D_reg_model.fit(features, Y)
+                coeffs[i1, i2, i3, :] = poly3D_reg_model.coef_
+                predicted = poly3D_reg_model.predict(features)
+                for idx in range(num_3D):
+                    idx_q1 = idx1[idx]
+                    idx_q2 = idx2[idx]
+                    idx_q3 = idx3[idx]
+                    predicted_3D[i1, i2, i3, idx_q1, idx_q2, idx_q3] = predicted[idx]
+
+    return coeffs, predicted_3D
+
+
+def taylor_coeffs(pes, deg=4, min_deg=3):
```

Comment:
I am a bit confused, what do you mean by "level" of Hamiltonian? Like the 4th order vs 6th order etc...?

Reply:
I mean are we doing till 1-mode pes, 2-mode pes or 3-mode pes for Hamiltonian generation.

Reply:
Oh I see. The current code allows you to do this without the need for a keyword, since you can just slice the output of `taylor_coeffs` and build your hamiltonian that way. I don't think it's useful to have this keyword -- we should assume the user, if they're working with taylor hamiltonians and etc, knows they can slice the taylor_coeffs array and build the hamiltonian that way instead.

Reply:
That way a user will be calculating the coefficients even if they don't need it.

Reply:
2) Is also the way you could do this without going into lower level functions such as taylor_coeffs. Starting with taylor_hamiltonian, you would just pass a custom 2-mode PES that was built using your 3-mode PES.